spec.emu

```diff
@@ -4,20 +4,1743 @@
+      <p>An object is a <dfn id="module-namespace-exotic-object" variants="module namespace exotic objects">module namespace exotic object</dfn> if its [[GetPrototypeOf]], [[SetPrototypeOf]], [[IsExtensible]], [[PreventExtensions]], [[GetOwnProperty]], [[DefineOwnProperty]], [[HasProperty]], [[Get]], [[Set]], [[Delete]], and [[OwnPropertyKeys]] internal methods use the definitions in this section, and its other essential internal methods use the definitions found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed by ModuleNamespaceCreate.</p>
+      <p>Module namespace exotic objects have the internal slots defined in <emu-xref href="#table-internal-slots-of-module-namespace-exotic-objects"></emu-xref>.</p>
+      <emu-table id="table-internal-slots-of-module-namespace-exotic-objects" caption="Internal Slots of Module Namespace Exotic Objects" oldids="table-29">
+        <table>
+          <tr>
+            <th>
+              Internal Slot
+            </th>
+            <th>
+              Type
+            </th>
+            <th>
+              Description
+            </th>
+          </tr>
+          <tr>
+            <td>
+              [[Module]]
+            </td>
+            <td>
+              a Module Record
+            </td>
+            <td>
+              The Module Record whose exports this namespace exposes.
+            </td>
+          </tr>
+          <tr>
+            <td>
+              [[Exports]]
+            </td>
+            <td>
+              a List of Strings
+            </td>
+            <td>
+              A List whose elements are the String values of the exported names exposed as own properties of this object. The list is ordered as if an Array of those String values had been sorted using %Array.prototype.sort% using *undefined* as _comparefn_.
+            </td>
+          </tr>
+        </table>
+      </emu-table>
+
+      <emu-clause id="sec-module-namespace-exotic-objects-get-p-receiver" type="internal method" number="8">
+        <h1>
+          [[Get]] (
+            _P_: a property key,
+            _Receiver_: an ECMAScript language value,
+          ): either a normal completion containing an ECMAScript language value or a throw completion
+        </h1>
+        <dl class="header">
+          <dt>for</dt>
+          <dd>a module namespace exotic object _O_</dd>
+        </dl>
+        <emu-alg>
+          1. If _P_ is a Symbol, then
+            1. Return ! OrdinaryGet(_O_, _P_, _Receiver_).
+          1. Let _exports_ be _O_.[[Exports]].
+          1. If _exports_ does not contain _P_, return *undefined*.
+          1. Let _m_ be _O_.[[Module]].
+          1. <ins>If _m_ is not a Cyclic Module Record, or both _m_.[[Status]] is ~linked~ and AnyDependencyNeedsAsyncEvaluation(_m_) is *false*, then</ins>
+            1. <ins>Let _promise_ be _m_.Evaluate().</ins>
```

Comment:
Could we look into refactoring the evaluation function here to skip creating a promise in the sync case. Alternatively could we explicitly craft an `EvaluteSync` function?

Reply:
In addition, such an `EvaluateSync` could itself inline the implementation of `AnyDependencyNeedsAsyncEvaluation`.